include/ht_refcounted.h

```diff
@@ -93,9 +94,11 @@ namespace Hatchit
             Handle(VarType* varPtr, uint32_t* refCounter, const Guid* name);
 
             //Private members
-            VarType* m_ptr;
-            uint32_t* m_refCount;
+            VarType*    m_ptr;
+            uint32_t*   m_refCount;
             const Guid* m_ID;
+
+            mutable std::mutex m_mutex;
```

Comment:
Shouldn't the actual mutex be with the refcounted object?  If two separate handles access the same object and have different mutexes, then we still have a race condition.